### PR TITLE
Fix .pc.in files to build in a snap

### PR DESCRIPTION
**Pull request by wmww**
This allowed me to build in a snap (it needs to be able to mangle $prefix), but I doubt it's actually correct. Please advise.

Edit: No rush, because the thing I actually care about it building my snap, which I'm already doing from the `mate-panel` branch which has this.

### CODE REVIEW

File: src/miral/CMakeLists.txt [hunk 1]
```diff
@@ -153,5 +154,5 @@ if(TARGET doc)
 endif()
 
 install(TARGETS     miral                           LIBRARY DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}")
-install(DIRECTORY   ${CMAKE_SOURCE_DIR}/include/miral       DESTINATION "${INCLUDEDIR}")
+install(DIRECTORY   ${CMAKE_SOURCE_DIR}/include/miral       DESTINATION "${PREFIX}/${INCLUDEDIR}")
```
Comment (Saviq) [inline src/miral/CMakeLists.txt]: This could use ${CMAKE_INSTALL_FULL_INCLUDEDIR}, no?

File: src/miral/CMakeLists.txt [hunk 2]
```diff
@@ -153,5 +154,5 @@ if(TARGET doc)
...
 install(FILES       ${CMAKE_CURRENT_BINARY_DIR}/miral.pc    DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig")
```
Comment (Saviq) [inline src/miral/CMakeLists.txt]: This could be simplified with `LIBRARY`?

**Comment by AlanGriffiths** [comment]
@wmww do we still need this PR?

**Comment by wmww** [comment]
Probably. The commit commit is still in the mate-waylnd snap branch, I just haven't felt motivated to get merged. I'll take another look and update this PR soon.

**Comment by AlanGriffiths** [comment]
Is this the problem you're seeing? https://forum.snapcraft.io/t/cmake-part-with-build-snaps-gets-mangled-include-parameters/12042

**Comment by wmww** [comment]
After almost 2 years, this is once again ready for review.

File: src/platform/mirplatform.pc.in [hunk 3]
```diff
@@ -1,11 +1,10 @@
...
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: mirplatform
 Description: Mir platform library
 Version: @MIR_VERSION@
 Requires: mircommon, mircore
-Libs: -L@LIBDIR@ -lmirplatform
-Cflags: -I@INCLUDEDIR@ -I@COMMON_INCLUDEDIR@
+Libs: -L${libdir} -lmirplatform
```
Comment (AlanGriffiths) [inline src/platform/mirplatform.pc.in:9]: I've not looked to see where the problem arises, but with this a local install gets a wrong mirplatform package config:
```plain
$ pkg-config --cflags mirplatform
-I/platform -I/common -I/usr/local/include/mircommon -I/usr/local/include/mircore
$ cat /usr/local/lib/pkgconfig/mirplatform.pc
prefix=/usr/local
libdir=/lib
includedir=/include
```

File: src/renderer/mirrenderer.pc.in [hunk 4]
```diff
@@ -1,10 +1,8 @@
...
-libdir=@LIBDIR@
-includedir=@INCLUDEDIR@
+prefix=@CMAKE_INSTALL_PREFIX@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: mirrenderer
 Description: Mir renderer
 Version: @MIR_VERSION@
-Cflags: -I@INCLUDEDIR@ -I@PLATFORM_INCLUDEDIR@ -I@COMMON_INCLUDEDIR@
+Cflags: -I{includedir}/mirrenderer -I{includedir}/mirplatform -I{includedir}/mircommon
```
Comment (AlanGriffiths) [inline src/renderer/mirrenderer.pc.in:7]:
```plain
$ cat /usr/local/lib/pkgconfig/mirrenderer.pc 
prefix=/usr/local
includedir=/include

Name: mirrenderer
Description: Mir renderer
Version: 2.0.0
Cflags: -I{includedir}/mirrenderer -I{includedir}/mirplatform -I{includedir}/mircommon
```

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
Please check all the .pc files get sane content. (I found a couple in error, and some correct but didn't check everything.)

File: src/platform/mirplatform.pc.in (hunk 3 above)
Comment (wmww) [reply src/platform/mirplatform.pc.in:9]: Ah! CMake screws it up unless `@ONLY` is passed to `configure_file().`

**Comment by wmww** [comment]
I think I fixed all the issues, and I looked at every .pc file in the build directory without noticing any obvious problems (`./tests/mirtest.pc` is a little strange, but that appears to be auto-generated and isn't effected by this PR)

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
```plain
$ pkg-config --cflags mirrenderer 
-I\{includedir\}/mirrenderer -I\{includedir\}/mirplatform -I\{includedir\}/mircommon
$ cat /usr/local/lib/pkgconfig/mirrenderer.pc 
prefix=/usr/local
includedir=${prefix}/include

Name: mirrenderer
Description: Mir renderer
Version: 2.0.0
Cflags: -I{includedir}/mirrenderer -I{includedir}/mirplatform -I{includedir}/mircommon
```

**Comment by wmww** [comment]
Ugh, sorry. Missed that. I've now sanity checked the result of running `pkg-config` with all the *.pc files, so maybe we're good now?

**Review by AlanGriffiths — APPROVED** [review]
LGTM

bors r+

**Pull request merged**